### PR TITLE
Refactoring Task Configurations and Adding Tests

### PR DESCRIPTION
**Pull request by awaemmanuel**
1. Refactor the Task Configurations from Yaml to OO objects
1. Add test cases for the refactor in tests folder
1. Test have passed

### CODE REVIEW

**Comment by awaemmanuel** [comment]
@chenhuims / @namisan - rebased to the new development branch and all tests have passed.  Feel free to review and comment.

**Review by namisan — APPROVED** [review]
It looks good to me.

File: tests/config/test_task_config.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+)
+
+
+@pytest.fixture(scope="function")
+def test_object():
+    """Fixture to create test config objects.
+    This method sets up a fixture that accepts a request object into a test function
+    """
+    print("setUp: test squad object creation")
+    data = {
```
Comment (chenhuims) [inline tests/config/test_task_config.py:39]: It looks that `data` is not used in any test function. Is this necessary?

**Review by chenhuims — COMMENTED** [review]
Thanks for this PR. Looks great to me as well. I've only left a small comment.

**Pull request merged**